lightning-macros/src/lib.rs

```diff
@@ -74,3 +77,175 @@ pub fn maybe_await(expr: TokenStream) -> TokenStream {
+							new_fields.extend(trailing_dots.into_iter().map(token_to_stream));
+							assert!(fields_stream.peek().is_none());
+							break;
+						}
+
+						let ref_ident = fields_stream.next().unwrap();
+						expect_ident(&ref_ident, Some("ref"));
+						let field_name_ident = fields_stream.next().unwrap();
+						let co = fields_stream.next().unwrap();
+						expect_punct(&co, ':');
+						let ty_info = fields_stream.next().unwrap();
+						let com = fields_stream.next().unwrap();
+						expect_punct(&com, ',');
+
+						if let TokenTree::Group(group) = ty_info {
+							let first_group_tok = group.stream().into_iter().next().unwrap();
+							if let TokenTree::Ident(ident) = first_group_tok {
+								if ident.to_string() == "legacy" {
+									continue;
+								}
+							}
+						}
+
+						let field = [ref_ident, field_name_ident, com];
+						new_fields.extend(field.into_iter().map(token_to_stream));
+					}
+					let fields_group = Group::new(Delimiter::Brace, new_fields);
+					computed_fields.extend(token_to_stream(TokenTree::Group(fields_group)));
+				} else {
+					computed_fields.extend(token_to_stream(TokenTree::Group(group)));
+				}
+			}
+
+			let arm_pfx = [enum_ident, co1, co2, variant_ident];
+			new_arms.extend(arm_pfx.into_iter().map(token_to_stream));
+			new_arms.extend(computed_fields);
+			let arm_sfx = [eq, gt, init];
+			new_arms.extend(arm_sfx.into_iter().map(token_to_stream));
+		}
+
+		let new_arm_group = Group::new(Delimiter::Brace, new_arms);
+		res.extend(token_to_stream(TokenTree::Group(new_arm_group)));
+	} else {
+		panic!("Expected `match self {{..}}` and nothing else");
+	}
+
+	assert!(stream.next().is_none(), "Expected `match self {{..}}` and nothing else");
+
+	res
+}
+
+/// Scans an enum definition for fields initialized to `LDK_DROP_LEGACY_FIELD_DEFINITION` and drops
+/// that field.
+#[proc_macro]
+pub fn drop_legacy_field_definition(expr: TokenStream) -> TokenStream {
+	let mut st = if let Ok(parsed) = parse::<syn::Expr>(expr) {
+		if let syn::Expr::Struct(st) = parsed {
+			st
+		} else {
+			return (quote! {
+				compile_error!("drop_legacy_field_definitions!() can only be used on struct expressions")
+			})
+			.into();
+		}
+	} else {
+		return (quote! {
+			compile_error!("drop_legacy_field_definitions!() can only be used on expressions")
+		})
+		.into();
+	};
+	assert!(st.attrs.is_empty());
+	assert!(st.qself.is_none());
+	assert!(st.dot2_token.is_none());
+	assert!(st.rest.is_none());
+	let mut new_fields = syn::punctuated::Punctuated::new();
+	core::mem::swap(&mut new_fields, &mut st.fields);
+	for field in new_fields {
+		if let syn::Expr::Macro(syn::ExprMacro { mac, .. }) = &field.expr {
+			let macro_name = mac.path.segments.last().unwrap().ident.to_string();
+			let is_init = macro_name == "_init_tlv_based_struct_field";
+			let ty_tokens = mac.tokens.clone().into_iter().skip(2).next();
+			if let Some(proc_macro2::TokenTree::Group(group)) = ty_tokens {
```

Comment:
From what I understand, the input field for this would look something like:
```rust

field: _init_tlv_based_struct_field!(field_name, (legacy, ...))
````
since the second element needs to be a group. I’m a bit unsure, though, about what exactly should go in place of `...`—I’d love any insights on that!
Also, maybe it would be helpful to expand the docs a bit to clearly outline the expected input and behavior of the macro for future reference.